apps/spark/src/main/java/com/linkedin/openhouse/jobs/scheduler/JobsScheduler.java

```diff
@@ -173,31 +176,33 @@ protected void run(
       try {
         long passedTimeMillis = System.currentTimeMillis() - startTimeMillis;
         long remainingTimeMillis = TimeUnit.HOURS.toMillis(tasksWaitHours) - passedTimeMillis;
-        if (remainingTimeMillis <= 0) {
-          // treat as a global timeout case similar to future.get timeout
-          throw new TimeoutException();
-        }
         jobState = taskFuture.get(remainingTimeMillis, TimeUnit.MILLISECONDS);
```

Comment:
Does taskFuture.get with negative value returns TimeoutException or other exception? Better to verify if it waits longer. Usually positive value is provided.

Reply:
Given negative value, it will return jobState if it is exists, otherwise TimeoutException.

Reply:
Ok nice.